### DISCUSSION TITLE
Secrets in ENV Variables

### DISCUSSION THREAD

**Discussion by gaokevin1** [vercel/community#3484] [Feature Question] [answered]
If I have API keys or some kind of secrets that need to be stored as a Vercel environment variable, I have no way of hiding these if someone else has access to my Vercel account. Secrets that are generated, like API and Management Keys, are usually treated as one-time viewed keys that cannot be viewed again. 

Is there a way we can differentiate in the Vercel, and if I use an API key or some other secret, it won't be displayed again? I don't believe this is currently offered as a feature, so a way to handle this would be great.

**Comment by leerob** [comment]
> I have no way of hiding these if someone else has access to my Vercel account

Are you saying that multiple people are logging into your personal Vercel account? Or referencing a team? If the former, that doesn't sound intended – your account is your own account 😄 For the later, you have different roles (e.g RBAC) on teams that allow you to scope access to who can [manage env variables](https://vercel.com/docs/accounts/team-members-and-roles/access-roles).

> Is there a way we can differentiate in the Vercel, and if I use an API key or some other secret, it won't be displayed again? I don't believe this is currently offered as a feature, so a way to handle this would be great.

We currently offer the ability to have secrets (non-viewable after generated) as environment variables for Enterprise customers. We'll continue to monitor feedback here and consider is this makes sense for Pro teams as well 🙏

**Reply by schulzfel** [reply]
IMO it makes total sense even in the free tier, it's just best practice. In our company we need to use env variables for financial services api keys, if they leak money we're ruined... 
If you have the feature ready for enterprise users, I don't understand why not making it generally available

**Reply by biwano** [reply]
Yeah I agree, it completly makes sense. Secrets are really a core feature for any automated deployment system.

**Comment by leerob** [comment, marked as answer]
We now have the ability to mark environment variables as sensitive!

https://vercel.com/changelog/sensitive-environment-variables-are-now-available
